### PR TITLE
Added RSA Algorithm Explainer

### LINKED ISSUE #179: [FEATURE] RSA Algorithm Explanation
### Detailed description

Add a explainer for RSA Algorithm with Example and Code Snippets.

### Context

This change should introduce a explainer for RSA Algorithm for encryption and decryption.

### Possible implementation

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by TanmoySG**
In this PR,

- Added Explainer for RSA Encryption and Decryption
- Added Block Diagrams 
- Added Example for Explaining the Encryption and Decryption
- Added Simple Python Implementation Code

Fixes #179

### CODE REVIEW

**Comment by TanmoySG** [comment]
Hey @vbrazo & @Panquesito7 ! Needed some help with this PR. Could you please review it, let me know if any change in required?

File: en/Ciphers/rsa_cipher.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    Enter the Cipher Text: 116
+    Enter your Private Key: 23
+    PlainText:  51
+    
+
+```
+
+## Implementations
+- [Python](https://github.com/TheAlgorithms/Python/blob/master/ciphers/rsa_cipher.py)
+- [Golang](https://github.com/TheAlgorithms/Go/blob/master/cipher/rsa/rsa.go)
```
Comment (Panquesito7) [inline en/Ciphers/rsa_cipher.md:159]:
```suggestion
## Implementations

- [Python](https://github.com/TheAlgorithms/Python/blob/master/ciphers/rsa_cipher.py)
- [GoLang](https://github.com/TheAlgorithms/Go/blob/master/cipher/rsa/rsa.go)
- [Java](https://github.com/TheAlgorithms/Java/blob/master/src/main/java/com/thealgorithms/ciphers/RSA.java)
- [Ruby](https://github.com/TheAlgorithms/Ruby/blob/master/ciphers/rsa.rb)
```

**Review by Panquesito7 — COMMENTED** [review]
Nice work! 🚀

**Review by Panquesito7 — DISMISSED** [review]
Great work! Thank you. 🚀
CC: @appgurueu.

File: en/Ciphers/rsa_cipher.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+# RSA Cryptography Algorithm
+
+## About RSA Algorithm
+RSA (Rivest–Shamir–Adleman) is one of the first public-key cryptosystems and is widely used for secure data transmission. In such a cryptosystem, the encryption key is public and distinct from the decryption key which is kept secret (private).
+
+## Algorithm
+
+1. Select 2 Prime Numbers - **p & q**
+2. Calculate `n` as $$n = p * q$$
+3. Calculate Euler's Totient Function of n, $$φ(n) = (p-1) * (q-1)$$
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:10]: Perhaps make it clear that this formula for Euler's Totient is only applicable **iff** `p` and `q` are prime. Also consider explaining (briefly) what Euler's Totient *is*.

File: en/Ciphers/rsa_cipher.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+4. Select PUBLIC KEY - $e$ , such that $e$ and $φ(n)$ are Co-primes, i.e, $$\gcd(e , φ(n))=1$$
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:11]: Formatting-wise, please use **bold** instead of CAPS consistently.

File: en/Ciphers/rsa_cipher.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+5. Calculate PRIVATE KEY, $d$ such that $$(d * e) \mod φ(n) = 1$$
+
+## Public and Private Keys
+
+1. The **Public  key** is `{ e , n }` which is known to all in the network.
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:16]: `{ ... }` is for sets, but this is a tuple?

File: en/Ciphers/rsa_cipher.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+The Plain Text, M is generated from the ciphertext, C using the private key, d as:
+
+$$M = C^d \mod n$$
+
+<!-- ![RSA-block-diagram%20%281%29.png](https://github.com/TanmoySG/Cryptography-Algorithm/blob/master/RSA-Algorithm/diagrams/RSA-block-diagram-image.png) -->
+
+## Encryption & Decryption Mechanism
+
+The Encryption and Decryption mechanishm block diagram with sample message encryption.
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:37]: Typo: mechanism*

File: en/Ciphers/rsa_cipher.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+print("CipherText: ", encrypt(pt))
+
+print(" ")
+
+#CipherText Input
+ct = int(input('Enter the Cipher Text: '))
+print("PlainText: ", decrypt(ct))
+```
+
+```shell
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:132]: This is not "shell", this is just the output.

File: en/Ciphers/rsa_cipher.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+![RSA-Example-maths-only-diagram.png](https://github.com/TanmoySG/Cryptography-Algorithm/blob/master/RSA-Algorithm/diagrams/RSA-Example-maths-only-diagram.png)
+
+## Simple Implementation of RSA using Python
+
+```python
+from sympy import *
+import math 
+
+#Generate p and q
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:53]: Please add a space after the `#` (for all comments)

File: en/Ciphers/rsa_cipher.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+p = randprime(1, 10)
+q = randprime(11, 20)
+
+#Generate n and l(n)
+n = p*q
+l = (p-1)*(q-1)
+
+#Function to test Co-Primality for generation of list of Public Keys
+def isCoPrime(x):
+    if math.gcd(l,x)==1:
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:63]: Just `return math.gcd(l, x) == 1`.

File: en/Ciphers/rsa_cipher.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+        return True
+    else:
+        return False
+
+#Function to find mod Inverese of e withl(n) to generate d     
+def modInverse(e, l) :
+    e = e % l;
+    for x in range(1, l) :
+        if ((e * x) % l == 1) :
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:72]: You don't need the surrounding parentheses here.

File: en/Ciphers/rsa_cipher.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+#Function to test Co-Primality for generation of list of Public Keys
+def isCoPrime(x):
+    if math.gcd(l,x)==1:
+        return True
+    else:
+        return False
+
+#Function to find mod Inverese of e withl(n) to generate d     
+def modInverse(e, l) :
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:69]: Please keep your formatting consistent. Don't add a space before the colon `:`.

File: en/Ciphers/rsa_cipher.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+    e = e % l;
```
Comment (appgurueu) [inline en/Ciphers/rsa_cipher.md:70]: You don't need semicolons in Python

**Review by appgurueu — COMMENTED** [review]
Good overall, but needs better formatting

File: en/Ciphers/rsa_cipher.md (hunk 2 above)
Comment (TanmoySG) [reply en/Ciphers/rsa_cipher.md]: Done

File: en/Ciphers/rsa_cipher.md (hunk 4 above)
Comment (TanmoySG) [reply en/Ciphers/rsa_cipher.md]: Done

File: en/Ciphers/rsa_cipher.md (hunk 7 above)
Comment (TanmoySG) [reply en/Ciphers/rsa_cipher.md:58]: Done

**Comment by TanmoySG** [comment]
Updated according to suggestions @appgurueu  @Panquesito7

**Review by appgurueu — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]
Awesome work! Thank you. 🚀

**Pull request merged**